src/main/java/frc/robot/hardware/MotorController.java

```diff
@@ -98,20 +100,22 @@ public boolean getReversed(){
 
     public static CANSparkMax constructMotor(MotorConfig config){
         CANSparkMax motor = new CANSparkMax(config.getID(), MotorType.kBrushless);
-        motor.restoreFactoryDefaults();
-        motor.setSmartCurrentLimit(config.getCurrentLimit());
-        motor.setIdleMode(config.getIdleMode());
-        motor.setOpenLoopRampRate(config.getOpenLoopRampRate());
+        int errorCode = motor.restoreFactoryDefaults().value;
+        errorCode += motor.setSmartCurrentLimit(config.getCurrentLimit()).value;
+        errorCode += motor.setIdleMode(config.getIdleMode()).value;
+        errorCode +=  motor.setOpenLoopRampRate(config.getOpenLoopRampRate()).value;
         motor.setInverted(config.getReversed());
+        if(errorCode != 0){DriverStation.reportError("An Error has occured in constructMotor(MotorConfig config)", null);}
         return motor;
     }
 
     public static CANSparkMax constructMotor(MotorConfig config, double[] PIDArray){
         CANSparkMax motor = constructMotor(config);
         SparkMaxPIDController motorPIDcontroller = motor.getPIDController();
-        motorPIDcontroller.setP(PIDArray[0]);
-        motorPIDcontroller.setI(PIDArray[1]);
-        motorPIDcontroller.setD(PIDArray[2]);
+        int errorCode = motorPIDcontroller.setP(PIDArray[0]).value; //Gets REVLibError code value (Is 0 if no error occured)
+        errorCode += motorPIDcontroller.setI(PIDArray[1]).value;
+        errorCode += motorPIDcontroller.setD(PIDArray[2]).value;
+        if(errorCode != 0){DriverStation.reportError("An Error has occured in constructMotor(MotorConfig config, double[] PIDArray)", null);}
```

Comment:
This doesn't give useful information.  We need to at least see which motor we are experiencing an error on.